projectional/src/main/java/jetbrains/jetpad/projectional/generic/CollectionEditor.java

```diff
@@ -28,12 +29,15 @@
   private List<ViewT> myViews;
   private Property<ItemT> myForDeletion;
   private boolean myCanCreateNew;
+  private Predicate<ViewT> myReplaceWithNewOnDel;
 
-  protected CollectionEditor(List<ItemT> items, List<ViewT> views, Property<ItemT> forDeletion, boolean canCreateNew) {
+  protected CollectionEditor(List<ItemT> items, List<ViewT> views, Property<ItemT> forDeletion, boolean canCreateNew,
+      Predicate<ViewT> replaceWithNewOnDel) {
```

Comment:
This is the condition against `ViewT` (`Cell`). Because `CollectionEditor` is generic, it can't check the `Cell` (view item) to be empty, it simply does not know anything about Cell type, that's why it requires a condition from outside.